library/src/main/java/com/timehop/stickyheadersrecyclerview/caching/HeaderViewCache.java

```diff
@@ -38,29 +38,32 @@ public View getHeader(RecyclerView parent, int position) {
         header.setLayoutParams(new ViewGroup.LayoutParams(
             ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT));
       }
-
-      int widthSpec;
-      int heightSpec;
-
-      if (mOrientationProvider.getOrientation(parent) == LinearLayoutManager.VERTICAL) {
-        widthSpec = View.MeasureSpec.makeMeasureSpec(parent.getWidth(), View.MeasureSpec.EXACTLY);
-        heightSpec = View.MeasureSpec.makeMeasureSpec(parent.getHeight(), View.MeasureSpec.UNSPECIFIED);
-      } else {
-        widthSpec = View.MeasureSpec.makeMeasureSpec(parent.getWidth(), View.MeasureSpec.UNSPECIFIED);
-        heightSpec = View.MeasureSpec.makeMeasureSpec(parent.getHeight(), View.MeasureSpec.EXACTLY);
-      }
-
-      int childWidth = ViewGroup.getChildMeasureSpec(widthSpec,
-          parent.getPaddingLeft() + parent.getPaddingRight(), header.getLayoutParams().width);
-      int childHeight = ViewGroup.getChildMeasureSpec(heightSpec,
-          parent.getPaddingTop() + parent.getPaddingBottom(), header.getLayoutParams().height);
-      header.measure(childWidth, childHeight);
-      header.layout(0, 0, header.getMeasuredWidth(), header.getMeasuredHeight());
       mHeaderViews.put(headerId, header);
     }
+    measureHeader(parent, header);
     return header;
   }
 
+  private void measureHeader(RecyclerView parent, View header){
```

Comment:
I remove this commit. It fixed size of header view while rotation changed. But it may decrease performance.